TELCODOCS-321 D/S Docs & RN: METAL-1 (MPINSTALL-72) Metal Day 1 Networking (#42448)
openshift/openshift-docs: modules/ipi-install-creating-an-rhcos-images-cache.adoc

[id="ipi-install-creating-an-rhcos-images-cache_{context}"]
= (Optional) Creating an {op-system} images cache

> To employ image caching, you must download the {op-system-first} image used by the bootstrap VM to provision the different nodes. Image caching is optional, but especially useful when running the installer on a network with limited bandwidth.

Reviewer: SSG: s/installer/installation program 
https://redhat-documentation.github.io/supplementary-style-guide/#_i

Author: Updated as suggested.

> To employ image caching, you must download the {op-system-first} image used by the bootstrap VM to provision the different nodes. Image caching is optional, but especially useful when running the installer on a network with limited bandwidth.

Reviewer 2: ```suggestion
To employ image caching, you must download the {op-system-first} image used by the bootstrap VM to provision the different nodes. Image caching is optional but especially useful when running the installation program on a network with limited bandwidth.
```

Author: Updated as suggested.

[NOTE]
====

> The installer no longer needs the `clusterOSImage` {op-system} image, as the correct image is in the release payload.

Reviewer: s/installer/installation program

Author: Updated as suggested.

> The installer no longer needs the `clusterOSImage` {op-system} image, as the correct image is in the release payload.

Reviewer 2: ```suggestion
The installer no longer needs the `clusterOSImage` {op-system} image because the correct image is in the release payload.
```

Author: Updated as suggested.

> The {op-system} image, `clusterOSImage`, used by the installer is no longer needed, as the correct ISO image is in the release payload.

Author: Updated as suggested for clarity and to remove "ISO".

$ export COMMIT_ID=$(/usr/local/bin/openshift-baremetal-install version | grep '^built from commit' | awk '{print $4}')
----
+

> The ID determines which image the installer needs to download.

Reviewer 2: ```suggestion
The ID determines which image the installer must download.
```

> ! need to verb
Use "need to" only when "must" is too restrictive.

Author: Updated as suggested.

$ export RHCOS_QEMU_SHA_UNCOMPRESSED=$(curl -s -S https://raw.githubusercontent.com/openshift/installer/$COMMIT_ID/data/data/rhcos.json  | jq -r '.images.qemu["uncompressed-sha256"]')
----

. Download the image and place it in the `/home/kni/rhcos_image_cache` directory:
+
[source,terminal]
----
$ curl -L ${RHCOS_PATH}${RHCOS_QEMU_URI} -o /home/kni/rhcos_image_cache/${RHCOS_QEMU_URI}
----

> . Confirm SELinux type is of `httpd_sys_content_t` for the newly created file:

Author: Updated as suggested.

+
[source,terminal]
----
$ podman run -d --name rhcos_image_cache \ <1>
-v /home/kni/rhcos_image_cache:/var/www/html \
-p 8080:8080/tcp \
registry.centos.org/centos/httpd-24-centos7:latest
----
ifndef::upstream[]
+

> <1> Creates a caching webserver with the name rhcos_image_cache. This pod serves the bootstrapOSImage image in the install-config.yaml file for deployment.  

Reviewer: ```suggestion
<1> Creates a caching webserver with the name `rhcos_image_cache`. This pod serves the `bootstrapOSImage` image in the `install-config.yaml` file for deployment.  
```

Author: Updated as suggested.

----
ifndef::upstream[]
+

> The above command creates a caching webserver with the name `rhcos_image_cache`. This pod serves the `bootstrapOSImage` image in the `install-config.yaml` file, `${RHCOS_PATH}${RHCOS_QEMU_URI}?sha256=${RHCOS_QEMU_SHA_UNCOMPRESSED}`, for deployment.  

Reviewer 3: Can you put this sentence in a callout and remove the "above command" part? 
> <1> Creates a caching webserver with the name `rhcos_image_cache`. This pod serves the `bootstrapOSImage` image in the `install-config.yaml` file, `${RHCOS_PATH}${RHCOS_QEMU_URI}?sha256=${RHCOS_QEMU_SHA_UNCOMPRESSED}`,  for deployment. 

This is only my perspective so feel free to ignore, but the second sentence is hard to read because of the env variables are pushing the "for deployment" at the end of it, making it feel disconnected. If it makes sense, something like:
> <1> Creates a caching webserver with the name `rhcos_image_cache`. This pod serves the `bootstrapOSImage` image in the `install-config.yaml` file for deployment.

I'd try to add another sentence about the variables if they **have** to be included here.

Author: Updated as suggested.
